linode/manager
# Feature: [M3-6665] Dynamic Premium Plans support for Linodes creation flows

## Description 📝
This PR introduces a dynamic version of the Premium Plans support. It essentially achieves a few things:
- Improve messaging and information about available premium regions
- Prevents the user from attempting to create a Linode with premium plan in a non-eligible region (which would result in a frustrating API error)
- It also improves the TextTooltip component and its coverage.

The diff is a bit large due to prop drilling and test updates (new props).

**Note**: A similar treatment for GPU will be added in a subsequent PR

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-06-15 at 3 28 07 PM](https://github.com/linode/manager/assets/130582365/05777c07-1ce3-4ad1-b531-c4d38e55a247) | ![Screenshot 2023-06-15 at 3 26 57 PM](https://github.com/linode/manager/assets/130582365/fc0be0dc-6d4c-4a00-919f-61efc1825eeb) |

## How to test 🧪

> ⚠️ **This PR affects ALL linode creation flows (including K8 and node pools). Each flow should be tested for consistency, on both desktop and mobile**

> ⚠️ **You will need the following flags on your admin account to be able to get the premium regions and features to populate**: `premium_dedicatedcpu`, `dc-internal` and `new-dc-testing-alpha`. Having those flags enabled on your admin account will ensure the provision of two Premium Plans regions needed for testing this PR: `Washington, DC (us-iad)` and `Paris, FR (fr-par)`

--------

The verification steps underneath should be performed on the following paths:
**Linode Create:**
- /linodes/create
- /linodes/create?type=One-Click
- /create?type=StackScripts
- /linodes/create?type=Images
- /linodes/create?type=Backups
- /create?type=Clone%20Linode

**Kubernetes Create**
- /kubernetes/create

Existing Cluster Add node pool (you need to test this with both a cluster in a non-eligible region and a cluster in a **Premium Plans regions**
- /kubernetes/clusters/[id]/summary

### Verification steps
- Confirm this behavior on the premium tab when no region is selected (available premium plans data result may vary if you have Chicago enabled, which I don't)
![Screenshot 2023-06-15 at 3 50 32 PM](https://github.com/linode/manager/assets/130582365/760e11fa-d309-4436-ac86-e0c16440a773)
- Confirm this behavior with a non eligible premium region (ex: Newark)
![Screenshot 2023-06-15 at 3 52 40 PM](https://github.com/linode/manager/assets/130582365/f93148e7-f1c1-4f1e-a76d-8b000ddb4d43)
- Confirm no message is displayed and form is enabled with a premium plans region (ex: Paris, fr)

Steps and UI may vary slightly depending on the creation flow (ex Backups) but the behavior should be consistent across all.
